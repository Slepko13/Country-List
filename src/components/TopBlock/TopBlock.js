import React from 'react';
import './TopBlock.scss';
import earth from "../../assets/images/happy_earth.png";
import PropTypes from "prop-types";


const TopBlock = ({ countryDetails}) => {
  if(!countryDetails)  return (
            <div className="TopBlock " >
                <div className="choose__country">
                    <div className="choose__image">
                        <img className="choose__earth" src={earth}  alt="happy earth"/>
                    </div>
                    <div className="choose__title">Choose a card :)</div>
                </div>
            </div>
            )
        return (
            <div className="TopBlock " >
                <div className="country">
                    <div className="country__flags"></div>
                    <div className="choose__image">
                        <img className="choose__earth " src={earth}  alt="happy earth"/>
                    </div>
                </div>
            </div>
        )
}

export default TopBlock;

TopBlock.propTypes ={
   countryDetails: PropTypes.bool
}