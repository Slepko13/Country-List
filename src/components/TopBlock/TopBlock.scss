.TopBlock {
  height: 226px;
  margin-bottom: 26px;
  @media screen and (min-width: 992px) {
    display: none;
  }
  .choose__country {
    text-align: center;
    .choose__image {
      .choose__earth {
        width: 133px;
        height: 133px;
      }
    }
    .choose__title  {
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 44px;
      line-height: 52px;
      color: #000000;
      @media screen and (max-width: 375px) {
        font-size: 32px;
      }
    }
  }
  .country {
    width: 100%;
    height: 100%;
    background-color: #FFFFFF;
    position: relative;
    display: flex;
    justify-content: center;
    .country__flags {
      position: absolute;
      left: 0;
      bottom: 37px;
      height: 100px;
      width: 100%;
      background: url("../../assets/images/flags.png") center center no-repeat;
      background-size: contain;
      @media screen and(min-width:425px){
        bottom: 7px;
        height: 180px;
      }
    }
    .choose__image {
      .choose__earth {
        width: 133px;
        height: 133px;
      }
    }
  }
}