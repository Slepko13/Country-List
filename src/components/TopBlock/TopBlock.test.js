import React from "react";
import TopBlock from "./TopBlock";

const setUp = (props) => shallow(<TopBlock {...props}/>)

describe("Top block component", () => {

    it("should render  with   props(snapshot)", () => {
        let component = shallow(<TopBlock id="Custom id"/>);
        expect(component).toMatchSnapshot();
    });

    it("should render without props(snapshot)", () => {
        let component = shallow(<TopBlock />);
        expect(component).toMatchSnapshot();
    });
});

