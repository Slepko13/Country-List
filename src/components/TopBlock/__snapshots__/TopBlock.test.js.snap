// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top block component should render  with   props(snapshot) 1`] = `
<div
  className="TopBlock "
>
  <div
    className="country"
  >
    <div
      className="country__flags"
    />
    <div
      className="choose__image"
    >
      <img
        alt="happy earth"
        className="choose__earth "
        src="happy_earth.png"
      />
    </div>
  </div>
</div>
`;

exports[`Top block component should render without props(snapshot) 1`] = `
<div
  className="TopBlock "
>
  <div
    className="choose__country"
  >
    <div
      className="choose__image"
    >
      <img
        alt="happy earth"
        className="choose__earth"
        src="happy_earth.png"
      />
    </div>
    <div
      className="choose__title"
    >
      Choose a card :)
    </div>
  </div>
</div>
`;
